### PR TITLE
Extend 'foreach' by a --filter-name=REGEX flag

### PR DESCRIPTION
**Pull request by t-wissmann**
This migrates the 'foreach' command to the ArgParse/CallOrComplete system which
makes it easy to add command line flags to the invocation syntax.
For this, the ArgParse class had to be extended slightly such that it
also support additional arguments after the positional arguments and
flags. In the 'foreach' example, those additional arguments specify the
nested command, i.e. the 'loop body'.

### CODE REVIEW

**Comment by t-wissmann** [comment]
The CI is fine. The only issue here is a2x, that randomly breaks (this happend already in previous PRs):
```
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam
navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0
"/etc/asciidoc/docbook-xsl/manpage.xsl"
"/home/runner/work/herbstluftwm/herbstluftwm/build-1017/doc/herbstclient.xml"
returned non-zero exit status 6
```

**Review by blarz — CHANGES_REQUESTED** [review]
Just a small doc string issue, otherwise looks!

File: src/argparse.cpp [hunk 1]
```diff
@@ -190,6 +193,69 @@ void ArgParse::command(CallOrComplete invocation, function<int(Output)> command)
     }
 }
 
+/**
+ * @brief Wrapper for commands that use argparse for the first
+ * arguments, but then do something more complicated with the not
+ * yet parsed tokens after the
```
[the remark this answers is not in the document]
Comment (t-wissmann) [reply src/argparse.cpp]: oups. after the " positional arguments and flags." :-)

[the remark this answers is not in the document]
Comment (blarz) [reply src/argparse.cpp]: Thanks :)

**Review by blarz — APPROVED** [review]
Looks good to me :)

**Pull request merged**